python/mxnet/model.py

```diff
@@ -480,19 +481,24 @@ def _is_data_arg(name):
         """Check if name is a data argument."""
         return name.endswith('data') or name.endswith('label')
 
-    def _init_params(self, input_shapes, overwrite=False):
+    def _init_params(self, input_shapes, input_types=None, overwrite=False):
         """Initialize weight parameters and auxiliary states"""
         arg_shapes, _, aux_shapes = self.symbol.infer_shape(**input_shapes)
         assert(arg_shapes is not None)
 
+        # currently, input_types is not passed in
+        if input_types is None:
+            input_types = {k : mx_real_t for k in input_shapes.keys()}
+        arg_types, _, aux_types = self.symbol.infer_type(**input_types)
```

Comment:
We'd better make sure that the change here does not affect other existing examples (like image classification and LSTM).